pkg/client/monitoring/v1alpha1/types.go

```diff
@@ -193,6 +193,24 @@ type ServiceMonitorSpec struct {
 	NamespaceSelector NamespaceSelector `json:"namespaceSelector,omitempty"`
 }
 
+// AlertingRuleFile defines alerting rules for a Prometheus instance
+// TODO: Should we go with v1 instead of v1alpha1 right away?
+type AlertingRuleFile struct {
```

Comment:
If `AlertingRuleFile` is used only to hold the serialized contents of an alerting rule file (the type for which is not represented in this API), how does `AlertingRuleFile` compare to using a `ConfigMap`?

Reply:
@ironcladlou My plan is to vendor the upstream Prometheus alerting rule struct. Thereby we would have full OpenAPI generated deep validation on creation time (instant feedback). What do you think?

Reply:
I still have a couple `TODO`s on my list :/

https://github.com/coreos/prometheus-operator/pull/1333/files#diff-c8710fb169ff41dfb58c4a26a2f7a755R210

Reply:
That makes sense to me- cool!